src/main/java/org/gridsuite/securityanalysis/server/dto/ContingencyResultDTO.java

```diff
@@ -36,4 +41,19 @@ public static ContingencyResultDTO toDto(ContingencyEntity contingency) {
             .subjectLimitViolations(subjectLimitViolations)
             .build();
     }
+
+    // each contingencyResultDto will return multiple line (one for each limitViolation)
+    public List<List<String>> toCsvRows(Map<String, String> translations) {
+        List<List<String>> csvRows = new ArrayList<>();
+        this.getSubjectLimitViolations().forEach(lm -> {
+            List<String> csvRow = new ArrayList<>();
+            csvRow.add(this.getContingency().getContingencyId());
+            csvRow.add(CsvExportUtils.translate(this.getContingency().getStatus(), translations));
+            csvRow.add(lm.getSubjectId());
+
+            csvRows.add(Stream.concat(csvRow.stream(), lm.getLimitViolation().toCsvRow(translations).stream()).toList());
```

Comment:
wouldn't it be more efficient to do
```java
            csvRow.addAll(lm.getLimitViolation().toCsvRow(translations));
            csvRows.add(csvRow);
```
?

Reply:
fixed, it's more efficient and easier to read